docs/reference/query-dsl/percolate-query.asciidoc

```diff
@@ -679,3 +684,11 @@ GET /_search
 
 NOTE: The above example assumes that there is a `query` field of type
 `percolator` in the mappings.
+
+Given the design of percolation, it often makes sense to use separate indices for the percolate queries and documents
+being percolated, as opposed to a single index as we do in examples. There are a few benefits to this approach:
+
+- Because percolate queries contain a different set of fields from the percolated documents, using two separate indices
+allows for fields to be stored in a denser, more efficient way.
+- Percolate queries do not scale in the same way as other queries, so percolation performance may benefit from using
+a different index configuration, like the number of primary shards.
```

Comment:
Probably, one more notable benefit here is independence for the document ids and percolate query ids.

Reply:
Thanks @psylone for taking a look. I am still getting up to speed on percolator, would you be able to elaborate on what you mean by independence here, and what advantage it has?

Reply:
For example, when we index documents, we support identifiers for them (yes, it's not necessary, but likely we do it).

So, imagine we have book documents with ids `1, 2, 3` and so on. If we will index percolate queries in the same index it's not clear if we will use an integer number as id. Thus, we need to use something different for percolate query ids. Probably, strings with the query name or add some prefix like `query-1`.

In case of different indices, we're free to use integer numbers as ids for both documents and percolate queries because their ids will not intersect even if they have the same value.

Reply:
I think that whether it's important to have different ID namespaces depends on the exact use case and other data modelling decisions. I'd prefer to keep the motivation in this section more general. I appreciate the PR suggestions though!